include/xtypes/Module.hpp

```diff
@@ -56,6 +56,27 @@ class Module
         return inner_[submodule];
     }
 
+    using ModuleVisitor = std::function<void(const Module& mod)>;
+
+    void for_each_submodule(
+            ModuleVisitor visitor,
+            const bool recursive=true) const
```

Comment:
```suggestion
            bool recursive = true) const
```
A primitive type passed as a copy doesn't need to be const, the original value will never change.